REGRESSION(284269@main) [CMake][WPE] Missing generation of inspector.gresource

lauromoura opened the pull request:
#### 1fe32d327a6cccf5436deca62af9ccc7d5c85307
<pre>
REGRESSION(284269@main) [CMake][WPE] Missing generation of inspector.gresource
<a href="https://bugs.webkit.org/show_bug.cgi?id=280561">https://bugs.webkit.org/show_bug.cgi?id=280561</a>

Reviewed by Adrian Perez de Castro.

Add a custom target to ensure the generation of `inspector.gresource`.

* Source/WebKit/InspectorGResources.cmake:

Canonical link: <a href="https://commits.webkit.org/284443@main">https://commits.webkit.org/284443@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/38583b4d38554a1c80759633a18c33d2e6ac3d9b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69396 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48796 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22069 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73479 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20554 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56597 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20405 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55191 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13655 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72462 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44514 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59913 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35669 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41180 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18931 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63122 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17688 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75190 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13378 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16910 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62859 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13417 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59996 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62765 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15396 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10784 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4397 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44600 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45674 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46869 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45415 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (commented):
This isn't ideal because it requires knowledge of the implementation details of `WEBKIT_BUILD_INSPECTOR_GRESOURCES`. Another option is to pass the WebKit target as an argument to the macro. (That would require also editing PlatformGTK.cmake. GTK doesn't actually need this dependency to be explicit, since it produces a source file instead, but it should be harmless to add.)

aperezdc commented on Source/WebKit/PlatformWPE.cmake:
```diff
@@ -536,6 +536,10 @@ endif ()
+add_custom_target(inspector_gresource_target
+    DEPENDS ${WebInspectorUI_DERIVED_SOURCES_DIR}/inspector.gresource
+)
+add_dependencies(WebKit inspector_gresource_target)
```
Why the `_target` suffix? We already know it's a target. Also, the `WebKit` target does not need `inspector.gresource` to be built, what you want is to add the `ALL` flag in the `add_custom_target()` call:

```suggestion
add_custom_target(inspector-resources
    DEPENDS ${WebInspectorUI_DERIVED_SOURCES_DIR}/inspector.gresource
    ALL
)
```

aperezdc replied:
If you don't want to assume knowledge of how the `WEBKIT_BUILD_INSPECTOR_GRESOURCES()` macro works, you can move the `add_custom_target()`call _into_ the macro itself. Making the custom target depend on `ALL` works fine for both the WPE and GTK ports&mdash;in the case of GTK, the source dependency will pull the custom command earlier in the build process as a dependency for the `WebKit` target anyway (as intended).

lauromoura replied:
Thanks for the `ALL`  suggestion. I just had to reorder it so `ALL` comes right after the target name. Patch updated.

aperezdc reviewed (changes requested):
I hit this issue myself today trying to use `ninja -C build install` for a plain CMake build, and I was about to send a similar patch. I have a comment tho, please check below!

lauromoura committed:
REGRESSION(284269@main) [CMake][WPE] Missing generation of inspector.gresource
https://bugs.webkit.org/show_bug.cgi?id=280561

Reviewed by Adrian Perez de Castro.

Add a custom target to ensure the generation of `inspector.gresource`.

* Source/WebKit/InspectorGResources.cmake:

Canonical link: https://commits.webkit.org/284443@main
